source/common/stats/thread_local_store.cc

```diff
@@ -137,6 +138,18 @@ void ThreadLocalStoreImpl::initializeThreading(Event::Dispatcher& main_thread_di
   tls_->set([](Event::Dispatcher&) -> ThreadLocal::ThreadLocalObjectSharedPtr {
     return std::make_shared<TlsCache>();
   });
+
+  // Initialize stats for each thread's dispatcher. This can't be done in the InitializeCb passed
+  // to `set` above, it has to be done as a separate step here, because initializing stats causes
+  // new TLS slots to be created and SlotImpl::set isn't reentrant.
+  auto worker_id = std::make_shared<std::atomic<uint32_t>>(0);
+  tls_->runOnAllThreads([this, &tls, worker_id] {
+    auto& dispatcher = tls.dispatcher();
+    std::string name = (&dispatcher == main_thread_dispatcher_)
```

Comment:
Please log worker ID on creations here. This will provide cheap correlation for debug at least.